src/application/service/CreateSharingCode.ts

```diff
@@ -0,0 +1,13 @@
+import { UserId } from '../../domain/model/user/UserId';
+import { v4 as uuidv4 } from 'uuid';
+import { SharingCode } from '../../domain/model/sharingCode/SharingCode';
+import { SharingCodeRepository } from '../../domain/model/sharingCode/SharingCodeRepository';
+
+export class CreateSharingCode {
+  constructor(private sharingCodeRepository: SharingCodeRepository) {}
+
+  public async withUserId(userId: string): Promise<SharingCode> {
+    const sharingCode = new SharingCode(uuidv4(), new UserId(userId));
```

Comment:
I would probably invert the userId and code here so that when we generate a new code we can let the uuidv4() be handled by the SharingCode class.

Reply:
Yeah, ok.  There is another case (reading from the DB) where we instantiate with both, but your way is slightly cleaner for this scenario.